pyfr/integrators/base.py

```diff
@@ -62,6 +62,27 @@ def __init__(self, backend, rallocs, mesh, initsoln, cfg):
         # Abort computation
         self.abort = False
 
+        # Smoothly step to target time in the last near_t steps
+        self.near_t = self.cfg.getint('solver-time-integrator', 'dt-switch', 10)
+        self._dt_far = cfg.getfloat('solver-time-integrator', 'dt')
+        self._dt_near = None
+
+    def adjust_step(self, t):
+
+        if self.tcurr + self.near_t*self._dt_far <= t:
```

Comment:
I think we need to investigate this carefully. For example lets say 
tcurr = 3.0
dt_far = 1.0
target t = 15.0
Then there shouldn't be any adjustments to the dt as we approach target t because we can get there by doing nothing, but the current algorithm fiddles with dt unnecessarily.

Reply:
The algorithm changes `self._dt` once to `self.dt_near` and once when less than a `self._dt_far` step away. By default, it uses `self.dt_far`. 
So between one target `t` and the next, the value of `self._dt` changes 3 times.

The same will happen for the example you gave. There may be two adjustments to `self._dt` as we approach to target t to fix machine precision errors.

Reply:
@semi-h I have tested a few cases in a [Google Colab notebook](https://colab.research.google.com/drive/1upThO_hbxr25ThikrYOAJa-IHNGjq-pD?usp=sharing)